src/video_core/renderer_vulkan/vk_blit_helper.h

```diff
@@ -38,6 +39,23 @@ class BlitHelper {
 private:
     vk::Pipeline MakeComputePipeline(vk::ShaderModule shader, vk::PipelineLayout layout);
     vk::Pipeline MakeDepthStencilBlitPipeline();
+    vk::Pipeline MakeFilterPipeline(vk::ShaderModule fragment_shader, vk::PipelineLayout layout);
+
+    void FilterAnime4K(Surface& surface, const VideoCore::TextureBlit& blit);
+    void FilterBicubic(Surface& surface, const VideoCore::TextureBlit& blit);
```

Comment:
[VK_IMG_filter_cubic](https://registry.khronos.org/vulkan/specs/latest/man/html/VK_IMG_filter_cubic.html) can possibly be utilized here to remove the need to use a full graphics pipeline in favor of a simpler `vkCmdBlitImage`-operation if this filter is available.

[A lot of mobile devices feature it](https://vulkan.gpuinfo.org/listdevicescoverage.php?extension=VK_IMG_filter_cubic&platform=android) in particular as well as [Snapdragon-based Windows-on-Arm laptops](https://vulkan.gpuinfo.org/listdevicescoverage.php?platform=windows&extension=VK_IMG_filter_cubic).